src/c-rbtree.h

```diff
@@ -58,7 +59,10 @@ typedef struct CRBTree CRBTree;
  * C_RBNODE_INIT.
  */
 struct CRBNode {
-        unsigned long __parent_and_flags;
+        union {
+                unsigned long __parent_and_flags;
+                alignas(4) char __dmmy_for_struct_alignment;
+        };
```

Comment:
```suggestion
        /* we require minimum of 4-byte alignment to store flags */
        alignas(4) unsigned long __parent_and_flags;
```

`alignas` never lessens the alignment requirement, so it is safe to use a lower-bound. I think using it directly with a comment is still better readable than the union, but I agree that this is not necessarily obvious. What do you think?

Reply:
I don't think that works:

```
../src/c-rbtree.h:62:34: error: ‘_Alignas’ specifiers cannot reduce alignment of ‘__parent_and_flags’
   62 |         alignas(4) unsigned long __parent_and_flags;
      |                                  ^~~~~~~~~~~~~~~~~~
```

Reply:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=65685

Reply:
@dvdhrm ping :)

Reply:
Gaah, bug-report is open since 2015, this is not gonna get fixed, is it?
With this in mind, I think I am fine with this change.